napari/layers/tracks/_track_utils.py

```diff
@@ -104,12 +121,12 @@ def data(self, data: Union[list, np.ndarray]):
         # convert data to a numpy array if it is not already one
         data = np.asarray(data)
 
+        # check check the formatting of the incoming track data
+        data = self._validate_track_data(data)
```

Comment:
Just so I understand the logic here - you removed the check for lexicographic ordering in `_validate_track_data`, since the data ends up being sorted later anyway? The performance boost comes from removing the extra (unnecessary) sorting in `_validate_track_data` and `build_tracks`?

Reply:
Yes, we got some performance boost by reducing the number of sort operations. `_validate_track_data` was only called after the data was sorted, therefore, it was always correct and unnecessary.

The original implementation only sorted when the layer was initialized, but that was changed in #1818 but the validation was kept the same.

`_fast_points_lookup` method also provides a good performance boost for long movies.